framework/src/actions/AutoCheckpointAction.C

```diff
@@ -0,0 +1,22 @@
+#include "AutoCheckpointAction.h"
+#include "FEProblem.h"
+
+registerMooseAction("MooseApp", AutoCheckpointAction, "auto_checkpoint_action");
+
+InputParameters
+AutoCheckpointAction::validParams()
+{
+  InputParameters params = Action::validParams();
+
+  return params;
+}
+
+AutoCheckpointAction::AutoCheckpointAction(const InputParameters & params) : Action(params) {}
+
+void
+AutoCheckpointAction::act()
+{
+  auto cp_params = _factory.getValidParams("Checkpoint");
+  cp_params.setParameters("should_output", false);
+  _problem->addOutput("Checkpoint", "autosave", cp_params);
```

Comment:
One more thing as I wrap up the review. Now that I've thought through this more, we are going to be introducing an "MPI All Reduce" at every single output step with this PR. I had hoped to avoid something like that, but I don't think it's possible since it's likely that only one process will receive a signal and you need all ranks to work together to output working checkpoints. We may need a flag to override adding this by default. Fortunately, this isn't too hard. We could register this Action with the `[Problem]` block or `[Application]` block to tell the system not to add this Checkpoint object in cases where performance is really sensitive. Worse yet. we really only need one of these checkpoint objects (at the very top of the hierarchy in a Multiapp simulation). It would be cleaner if we only add this to the "ultimate parent" (<- Yes, that's a method somewhere, MooseApp?).